src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7734,29 +7734,30 @@ private BoundExpression BindIndexerAccess(ExpressionSyntax node, BoundExpression
             LookupOptions lookupOptions = expr.Kind == BoundKind.BaseReference ? LookupOptions.UseBaseReferenceAccessibility : LookupOptions.Default;
             CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
             this.LookupMembersWithFallback(lookupResult, expr.Type, WellKnownMemberNames.Indexer, arity: 0, useSiteInfo: ref useSiteInfo, options: lookupOptions);
-            diagnostics.Add(node, useSiteInfo);
 
             // Store, rather than return, so that we can release resources.
             BoundExpression indexerAccessExpression;
 
             if (!lookupResult.IsMultiViable)
             {
-                if (TryBindIndexOrRangeIndexer(
+                if (TryBindIndexOrRangeIndexerFallback(
                     node,
                     expr,
                     analyzedArguments,
                     diagnostics,
-                    out var patternIndexerAccess))
+                    out var indexerFallbackAccess))
                 {
-                    indexerAccessExpression = patternIndexerAccess;
+                    indexerAccessExpression = indexerFallbackAccess;
```

Comment:
>indexerAccessExpression = indexerFallbackAccess; [](http://example.com/codeflow?start=20&length=48)

It  is not obvious why is it the right thing to drop useSiteInfo on this code path.